message.js

```diff
@@ -93,10 +93,7 @@ export default class MessageCall  {
       // The other side is calling a function here
       let func = this._functions[message.path];
       assert(func, "404 " + message.path + ": No such function registered");
-      let result = func(message.arg);
-      if (result instanceof Promise) {
-        result = await result;
-      }
+      let result = await func(message.arg);
```

Comment:
Please revert this change.

The result is not always a promise. In fact, in many use cases, very often, it's not. While `await` works on non-promises, so the code as you write it will work, it does have an effect. It creates a Promise and awaits it. This is slower, due to the Promise creation, and also changes execution order.
Please read the very educating https://stackoverflow.com/a/55263084/7154362 for details.

I suspected as such and that's why I was conservative and do the await only when necessary.

Reply:
This comment is on the wrong PR; it belongs on PR#1. But the result is actually usually a promise, since most of the callbacks are async functions, the two exceptions being the call to register a remote class and the (nonfunctional as yet) call to release the reference to a local object.

Reply:
OK, fair enough.